src/canvas-decodes.js

```diff
@@ -0,0 +1,493 @@
+          // Planted Forest
+          imgData[pixelPos] = 232;
+          imgData[pixelPos + 1] = 163;
+          imgData[pixelPos + 2] = 229;
+        } else if (imgData[pixelPos] === 11) {
+          // Non-forest Wetlands
+          imgData[pixelPos] = 39;
+          imgData[pixelPos + 1] = 137;
+          imgData[pixelPos + 2] = 212;
+        } else if (imgData[pixelPos] === 12) {
+          // Grassland
+          imgData[pixelPos] = 204;
+          imgData[pixelPos + 1] = 219;
+          imgData[pixelPos + 2] = 152;
+        } else if (imgData[pixelPos] === 13) {
+          // Other Non-forest Vegetation
+          imgData[pixelPos] = 138;
+          imgData[pixelPos + 1] = 184;
+          imgData[pixelPos + 2] = 75;
+        } else if (imgData[pixelPos] === 15) {
+          // Pasture
+          imgData[pixelPos] = 255;
+          imgData[pixelPos + 1] = 184;
+          imgData[pixelPos + 2] = 126;
+        } else if (imgData[pixelPos] === 18) {
+          // Agriculture
+          imgData[pixelPos] = 210;
+          imgData[pixelPos + 1] = 169;
+          imgData[pixelPos + 2] = 101;
+        } else if (imgData[pixelPos] === 21) {
+          // Pasture or Agriculture
+          imgData[pixelPos] = 232;
+          imgData[pixelPos + 1] = 176;
+          imgData[pixelPos + 2] = 113;
+        } else if (imgData[pixelPos] === 23) {
+          // Beaches and Dunes
+          imgData[pixelPos] = 221;
+          imgData[pixelPos + 1] = 126;
+          imgData[pixelPos + 2] = 107;
+        } else if (imgData[pixelPos] === 24) {
+          // Urban Infrastructure
+          imgData[pixelPos] = 233;
+          imgData[pixelPos + 1] = 70;
+          imgData[pixelPos + 2] = 43;
+        } else if (imgData[pixelPos] === 25) {
+          // Other Non-vegetated Area
+          imgData[pixelPos] = 255;
+          imgData[pixelPos + 1] = 153;
+          imgData[pixelPos + 2] = 255;
+        } else if (imgData[pixelPos] === 26) {
+          // Water Bodies
+          imgData[pixelPos] = 163;
+          imgData[pixelPos + 1] = 220;
+          imgData[pixelPos + 2] = 254;
+        } else if (imgData[pixelPos] === 27) {
+          // Unobserved
+          imgData[pixelPos] = 235;
+          imgData[pixelPos + 1] = 236;
+          imgData[pixelPos + 2] = 236;
+          imgData[pixelPos + 3] = 0;
+        } else if (
+          imgData[pixelPos] === 1 ||
+          imgData[pixelPos] === 2 ||
+          imgData[pixelPos] === 10 ||
+          imgData[pixelPos] === 14
+        ) {
+          // Unknown / No data
+          imgData[pixelPos] = 256;
+          imgData[pixelPos + 1] = 256;
+          imgData[pixelPos + 2] = 256;
+          imgData[pixelPos + 3] = 0;
+        } else {
+          imgData[pixelPos] = 256;
+          imgData[pixelPos + 1] = 256;
+          imgData[pixelPos + 2] = 256;
+          imgData[pixelPos + 3] = 0;
+        }
+      }
+    }
+  }
+};
+
+export default {
+  // GFW
+  '78747ea1-34a9-4aa7-b099-bdb8948200f4': decodes.treeCover,
```

Comment:
So here is a tricky bit. I though that maybe we could use a unique slug instead of ID so that we can have multiple layers using the same key?

Reply:
As we have different slugs between apps, the problem would be having this slug stored somewhere in the API. We should select someplace to put it... it could be inside the layerConfig.

Reply:
Agree with @mbarrenechea. I think we should save a param inside the `layerConfig`, like `decoder: 'treeCover'`, this way we remove the id's dependency which is very dangerous and volatile and will give us a headache in the future for sure.

@edbrett by slug, you mean the decoders or slugs coming from the layer? Thing is we cannot depend on something autogenerated by the API (like ids).

Reply:
I am saying the same as both of you I just didn't give enough detail. We need a unique slug for each decode which can be attached to a layerConfig to access it.